packages/cardano-services/src/Asset/DbSyncAssetProvider.ts

```diff
@@ -64,8 +65,18 @@ export class DbSyncAssetProvider extends DbSyncProvider implements AssetProvider
     const assetInfo: Asset.AssetInfo = { assetId, fingerprint, mintOrBurnCount, name, policyId, quantity };
 
     if (extraData?.history) await this.loadHistory(assetInfo);
-    if (extraData?.nftMetadata)
-      assetInfo.nftMetadata = await this.#dependencies.ntfMetadataService.getNftMetadata(assetInfo);
+    if (extraData?.nftMetadata) {
+      try {
+        // Use `null` instead of `undefined` when no nftMetadata is found / asset is not an NFT.
+        // Undefined means that there was a problem while querying and users can retry it.
+
+        // REVIEW: getNftMetadata does db-sync query or blockfrost query. Should we filter errros instead
+        // of catching all and transforming to undefined?
+        assetInfo.nftMetadata = (await this.#dependencies.ntfMetadataService.getNftMetadata(assetInfo)) || null;
```

Comment:
I think it would be better to use the [nullish coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator) operator here
- assetInfo.nftMetadata = (await this.#dependencies.ntfMetadataService.getNftMetadata(assetInfo)) || null;
+ assetInfo.nftMetadata = (await this.#dependencies.ntfMetadataService.getNftMetadata(assetInfo)) ?? null;